src/pages/login.tsx

```diff
@@ -0,0 +1,117 @@
+import { NextPage } from 'next'
+import Image from 'next/image'
+import { Auth } from 'firebase/auth'
+
+import { auth } from '../components/Firebase/init'
+import { useAuth } from '../context/AuthContext'
+
+import ReactDOM from 'react-dom'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faApple, faFacebook, faFacebookF, faGithub, faGoogle, faMicrosoft, faTwitter, faYahoo } from '@fortawesome/free-brands-svg-icons'
+
+const Login: NextPage = () => {
+  const { googleSignIn, logOut } = useAuth()
+
+  const googleIcon = <FontAwesomeIcon icon={faGoogle} />
+  const facebookIcon = <FontAwesomeIcon icon={faFacebookF} />
+  const githubIcon = <FontAwesomeIcon icon={faGithub} />
+  const twitterIcon = <FontAwesomeIcon icon={faTwitter} />
+  const appleIcon = <FontAwesomeIcon icon={faApple} />
+  const microsoftIcon = <FontAwesomeIcon icon={faMicrosoft} />
+  const yahooIcon = <FontAwesomeIcon icon={faYahoo} />
+
+  function handleGoogle(auth: Auth) {
+    googleSignIn?.(auth)
+  }
+
+  return (
+    <>
+      <main>
+        <title>Login</title>
+        <div className='m-auto max-w-sm p-10'>
+          <Image
+            src='/images/poops-logo.jpg'
+            width={36}
+            height={36}
+            layout='responsive'
+            alt='POOPS logo'
+          ></Image>
+        </div>
+
+        <div className='text-center text-xl font-bold'>Sign In</div>
+
+        <div className='m-auto grid h-1/3 w-1/2 max-w-xs space-y-4 p-5'>
```

Comment:
Maybe consider refactoring this page into components? The amount of nested HTML elements is gonna be hard to follow if other developers end up working on this page.

Reply:
Especially since it looks like your buttons contain very similar code.